docs/calculations.mdx

```diff
@@ -213,6 +226,63 @@ Providers should clearly and publicly document:
 
 ## Calculating Emissions
 
+Account for non-required fields:
+
+```
+  pct_mobile =
+      country ?
+      default_percent_mobile_by_country[country] / 100 :
+      default_percent_mobile / 100
+
+  average_embodied_emissions_gco2e_per_kb =
+      default_network_embodied_emissions_gco2e_per_kb['mobile'] x pct_mobile +
+      default_network_embodied_emissions_gco2e_per_kb['fixed'] x (1 - pct_mobile)
+
+  embodied_emissions_gco2e_per_kb =
+      network_type ?
+      default_network_embodied_emissions_gco2e_per_kb[network_type] :
+      average_embodied_emissions_gco2e_per_kb
+
+  average_usage_kwh_per_gb =
+      default_usage_kwh_per_gb['mobile'] x pct_mobile +
+      default_usage_kwh_per_gb['fixed'] x (1 - pct_mobile)
+
+  usage_kwh_per_gb =
+      network_type ?
+      default_usage_kwh_per_gb['network_type'] :
+      average_usage_kwh_per_gb
+
+  average_baseload_watts =
+      default_baseload_watts['mobile'] x pct_mobile +
+      default_baseload_watts['fixed'] x (1 - pct_mobile)
+
+  average_dynamic_watts_per_kbps =
+      default_dynamic_watts_per_kbps['mobile'] x pct_mobile +
+      default_dynamic_watts_per_kbps['fixed'] x (1 - pct_mobile)
+
+  baseload_watts =
+      network_type ?
+      default_baseload_watts[network_type] :
+      average_baseload_watts
+
+  dynamic_watts_per_kbps =
+      network_type ?
+      default_dynamic_watts_per_kbps[network_type] :
+      average_dynamic_watts_per_kbps
+
+  is_audio_creative =
+      creative_audio_data_transfer_bytes ||
+      creative_audio_duration_seconds ||
+      ad_format.audio_duration_seconds
+  default_channel =
+      is_audio_creative ?
+      audio :
+      default_channel_by_device[device_type]
+  channel = channel ? default_channel
+
+  device_type = device_type ? default_device_by_channel[channel]
```

Comment:
circular dependency when we don't have either

Reply:
mm I was hoping you wouldn't notice that. Updated with a function to make the logic more clear. Also put `channels` explicitly on the property object.